### PR TITLE
Refactor voltage level dialog

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**

### CODE REVIEW

File: src/components/dialogs/input-hooks.js [hunk 1]
```diff
@@ -424,74 +354,99 @@ export const useConnectivityValue = ({
...
                 onChange={(event, newValue) => {
                     handleChangeValue(newValue);
                 }}
-                options={Object.keys(values.object())}
-                getOptionLabel={(code) => values.get(code)}
+                size={'small'}
+                options={values}
+                getOptionLabel={getLabel}
+                {...(allowNewValue && {
+                    freeSolo: true,
```
Comment (etiennehomer) [inline src/components/dialogs/input-hooks.js:399]: False ?

File: src/components/dialogs/voltage-level-creation-dialog.js [hunk 2]
```diff
@@ -87,228 +218,67 @@ const VoltageLevelCreationDialog = ({
...
+        formProps: filledTextField,
+        Field: BusBarSection,
+    });
+
+    const [connections, connectionsField] = useExpandableValues({
+        id: 'connections',
+        labelAddValue: 'CreateLink',
+        validateItem: validateConnection,
+        inputForm: inputForm,
+        formProps: filledTextField,
```
Comment (etiennehomer) [inline src/components/dialogs/voltage-level-creation-dialog.js:274]: Remove or add it in usseExpandableValues

File: src/components/dialogs/input-hooks.js [hunk 3]
```diff
@@ -424,74 +354,99 @@ export const useConnectivityValue = ({
...
+                            });
+                        }
+                        return filtered;
+                    },
+                })}
                 renderInput={(props) => (
                     <TextField
-                        {...formProps}
-                        {...props}
                         variant="filled"
```
Comment (etiennehomer) [inline src/components/dialogs/input-hooks.js:416]: Removed. We get it from formProps

**Review by etiennehomer — CHANGES_REQUESTED** [review]

**Comment by etiennehomer** [comment]
Tests OK

File: src/components/dialogs/voltage-level-creation-dialog.js (hunk 2 above)
Comment (JacquesBorsenberger) [reply src/components/dialogs/voltage-level-creation-dialog.js]: done

**Review by laurgarn — APPROVED** [review]

File: src/components/dialogs/voltage-level-creation-dialog.js [hunk 4]
```diff
@@ -4,74 +4,205 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
-import React, { useState } from 'react';
+import React, { useEffect, useRef } from 'react';
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/dialogs/voltage-level-creation-dialog.js]: removed

File: src/components/dialogs/voltage-level-creation-dialog.js [hunk 5]
```diff
@@ -4,74 +4,205 @@
...
+        (m, v) => m.set(keyPosition(v), (m.get(keyPosition(v)) || 0) + 1),
+        new Map()
+    );
+    values.forEach((val, idx) => {
+        const errorId = idMap.get(val.id) > 1;
+        const errorPosition = posMap.get(keyPosition(val)) > 1;
+        if (errorId || errorPosition)
+            res.set(idx, {
+                error: true,
+                ...(errorId && { BusBarSectionID: 'not unique' }),
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/dialogs/voltage-level-creation-dialog.js]: not missing, just at wrong value, corrected

File: src/components/dialogs/input-hooks.js (hunk 3 above)
Comment (JacquesBorsenberger) [reply src/components/dialogs/input-hooks.js:417]: change reverted

Comment (etiennehomer) [reply src/components/dialogs/input-hooks.js:423]: Yes but variant="filled" has to be removed

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**